include/seqan/index/find2_index_approx.h

```diff
@@ -0,0 +1,585 @@
+                               uint8_t const blockIndex,
+                               TDir const & /**/,
+                               TDistanceTag const & /**/)
+{
+    bool goToRight2 = (blockIndex < s.pi.size() - 1) && s.pi[blockIndex + 1] > s.pi[blockIndex];
+    if (std::is_same<TDir, Rev>::value)
+    {
+        uint32_t infixPosLeft = needleRightIt - 1;
+        uint32_t infixPosRight = needleLeftIt + s.blocklength[blockIndex] - 1;
+
+        if (!goDown(iter, infix(needle, infixPosLeft, infixPosRight + 1), TDir()))
+            return;
+
+        if (goToRight2)
+            _schemeSearch(delegate, iter, needle, needleLeftIt, infixPosRight + 2, errors, s,
+                          std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1), Rev(), TDistanceTag());
+        else
+            _schemeSearch(delegate, iter, needle, needleLeftIt, infixPosRight + 2, errors, s,
+                          std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1), Fwd(), TDistanceTag());
+    }
+    else
+    {
+        // has to be signed, otherwise we run into troubles when checking for -1 >= 0u
+        int32_t infixPosLeft = needleRightIt - s.blocklength[blockIndex] - 1;
+        int32_t infixPosRight = needleLeftIt - 1;
+
+        while (infixPosRight >= infixPosLeft)
+        {
+            if (!goDown(iter, needle[infixPosRight], TDir()))
+                return;
+            --infixPosRight;
+        }
+        if (goToRight2)
+            _schemeSearch(delegate, iter, needle, infixPosLeft, needleRightIt, errors, s,
+                          std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1), Rev(), TDistanceTag());
+        else
+            _schemeSearch(delegate, iter, needle, infixPosLeft, needleRightIt, errors, s,
+                          std::min(blockIndex + 1, static_cast<uint8_t>(s.u.size()) - 1), Fwd(), TDistanceTag());
+    }
+}
+
+template <typename TDelegate, typename TText, typename TIndex, typename TIndexSpec, typename TText2, size_t nbrBlocks,
+          typename TDir, typename TDistanceTag>
+inline void _schemeSearch(TDelegate & delegate,
+                          Iter<Index<TText, BidirectionalIndex<TIndex> >, VSTree<TopDown<TIndexSpec> > > iter,
+                          TText2 const & needle,
+                          uint32_t const needleLeftIt,
+                          uint32_t const needleRightIt,
+                          uint8_t const errors,
+                          Search<nbrBlocks> const & s,
+                          uint8_t const blockIndex,
+                          TDir const & /**/,
+                          TDistanceTag const & /**/)
+{
+    uint8_t const maxErrorsLeftInBlock = s.u[blockIndex] - errors;
+    uint8_t const minErrorsLeftInBlock = (s.l[blockIndex] > errors) ? (s.l[blockIndex] - errors) : 0;
+
+    // Done.
+    if (minErrorsLeftInBlock == 0 && needleLeftIt == 0 && needleRightIt == length(needle) + 1)
+    {
```

Comment:
this OTOH doesn't need braces ;-)

Reply:
It does because the else-block needs braces ;-)

Reply:
haha, then it proves my next point